### ISSUE TITLE
INC0036853 Docker Image Registry full- SPIKE

### ISSUE THREAD

**Issue by sbarre-esit** [BCDevOps/developer-experience#499]
The platforms docker image registry has now hit 95% full. Here are the top 10 namespaces by usage.

wksv3k-tools is 272 GB
jik2hd-tools is 136 GB
empr-mds-tools is 124 GB
ifttgq-tools is 97 GB
devhub-tools is 95 GB
eazios-tools is 42 GB
q6qfzk-tools is 37 GB
lclb-cllc-tools is 37 GB
l9fjgg-tools is 34 GB
yuumcs-tools is 33 GB

It appears some projects make new tags for each PR build, and never clean them up. Leading to hundreds of tagged images that wont be pruned.

**Current definition for "done" regarding this card is auditing and addressing all projects consuming 100GB or more.**

**Comment by sbarre-esit** [comment]
@mitovskaol FYI I've started poking at some of the teams in the Ops channel, but perhaps we need to get this into the training/docs somewhere as well?

**Comment by mitovskaol** [comment]
@sbarre-esit Thanks for bringing this up. It looks like it is another "capacity" issue where we need the entire community to step in and work together to resolve. I asked @caggles to look into putting together another "self-help" doc for this.

**Comment by mitovskaol** [comment]
The teams have been contacted and asked to make changes to reduce their use of the docker image registry.... waiting to see the results of their actions.

**Comment by sbarre-esit** [comment]
wksv3k-tools is 183 GB
jik2hd-tools is 136 GB
empr-mds-tools is 123 GB
devhub-tools is 22 GB

/registry is now at 82% full. Will keep following up with jik2hd-tools and empr-mds-tools

**Comment by sbarre-esit** [comment]
Alec made a thing https://nbviewer.jupyter.org/github/bcgov/cas-notebooks/blob/master/ImageLayers.ipynb

**Comment by sbarre-esit** [comment]
Updated sizes of top 10 repos

wksv3k-tools is 165 GB
jik2hd-tools is 89 GB
empr-mds-tools is 130 GB
ifttgq-tools is 97 GB
devhub-tools is 23 GB
eazios-tools is 32 GB
q6qfzk-tools is 39 GB
lclb-cllc-tools is 4 GB
l9fjgg-tools is 34 GB
yuumcs-tools is 24 GB

**Comment by sbarre-esit** [comment]
Looks like deleting an ImageStream may not actually clean up the disk of the registry during soft prune?


```
$ oc -n jik2hd-tools get is -o name
imagestream.image.openshift.io/jenkins
imagestream.image.openshift.io/jenkins-basic
imagestream.image.openshift.io/jenkins-slave-main
imagestream.image.openshift.io/jh-etk-buildimage
imagestream.image.openshift.io/jh-etk-db-backup
imagestream.image.openshift.io/jh-etk-db-image
imagestream.image.openshift.io/jh-etk-image
imagestream.image.openshift.io/redhat-openjdk18-openshift
imagestream.image.openshift.io/rhel-atomic

$ oc rsh docker-registry-609-6lhfp
sh-4.2$ cd /registry/docker/registry/v2
sh-4.2$ ls repositories/jik2hd-tools/
jenkins             jh-etk-db-backup    jh-etk-icbcmocksvc  jh-etk-paymentsvc
jenkins-basic       jh-etk-db-image     jh-etk-image        jh-etk-primeadapter
jenkins-slave-main  jh-etk-disputesvc   jh-etk-issuancesvc  jh-etk-scweb
jh-etk-buildimage   jh-etk-icbcadapter  jh-etk-jiadapter    redhat-openjdk18-openshift
```

Hopefully the hard-prune will get those for us.

**Comment by mitovskaol** [comment]
Waiting for the hard-prune script to run on July 1 to see the results of the clean-up efforts.

**Comment by wmhutchison** [comment]
Followed up on checking the above queries for jik2hd-tools to see if hard prune garnered any results. Looks promising compared to before:

```
$ oc -n jik2hd-tools get is -o name
imagestream.image.openshift.io/jenkins
imagestream.image.openshift.io/jenkins-basic
imagestream.image.openshift.io/jenkins-slave-main
imagestream.image.openshift.io/jh-etk-buildimage
imagestream.image.openshift.io/jh-etk-db-backup
imagestream.image.openshift.io/jh-etk-db-image
imagestream.image.openshift.io/jh-etk-image
imagestream.image.openshift.io/redhat-openjdk18-openshift
imagestream.image.openshift.io/rhel-atomic

$ oc -n default rsh docker-registry-626-qjh2j
sh-4.2$ cd /registry/docker/registry/v2
sh-4.2$ ls repositories/jik2hd-tools/
jenkins  jenkins-basic  jenkins-slave-main  jh-etk-buildimage  jh-etk-db-backup  jh-etk-db-image  jh-etk-image  redhat-openjdk18-openshift
```

Running a fresh audit of the registry now to see who are the new heavy-hitters for disk space.

**Comment by wmhutchison** [comment]
New top 10:

ifttgq-tools is 105 GB
empr-mds-tools is 93 GB
jik2hd-tools is 89 GB
q6qfzk-tools is 44 GB
l9fjgg-tools is 34 GB
mem-tfrs-tools is 33 GB
vmvfjv-tools is 32 GB
gcpe-news-tools is 27 GB
frjeow-tools is 26 GB
dqszvc-dev is 26 GB

**Comment by wmhutchison** [comment]
Discussed internally with Steven - setting threshold for "Done" to be projects being under 100GB in consumption. Will create a new card after finishing this audit to create a weekly report of all projects over 100GB for ongoing follow-up.

**Comment by wmhutchison** [comment]
Refreshed the audit, going after the owners of the last two entries since they match the 100GB threshold.

sh-4.2$ sort -g /tmp/2020july14_results.txt | tail -10
30 GB is the size of wksv3k-tools
32 GB is the size of vmvfjv-tools
33 GB is the size of dqszvc-dev
33 GB is the size of mem-tfrs-tools
34 GB is the size of l9fjgg-tools
40 GB is the size of 8ecbmv-tools
45 GB is the size of q6qfzk-tools
76 GB is the size of jik2hd-tools
101 GB is the size of empr-mds-tools
105 GB is the size of ifttgq-tools

**Comment by lukegonis** [comment]
DOD: follow up with two product teams who's usage of image registry is above 100gb

**Comment by wmhutchison** [comment]
Finally got the following out on Rocket Chat for the current three over 100GB.

@rloisell @jennifer.dowd @parul.dubedy @garywong-bc Greetings - am reaching out as part of an ongoing audit of consumed registry disk space, and wish for you all to investigate the following projects you are responsible for, and please untag images no longer needed so they can be pruned. Thank you. :)

Project `jik2hd-tools` is consuming 105GB of raw registry space.
Project `ifttgq-tools` is consuming 108 GB of raw registry space.
Project `empr-mds-tools` is consuming 118GB of raw registry space.

**Comment by wmhutchison** [comment]
Other fun facts derived from the output of the raw registry disk audit.

223GB consumed by projects with less than 1GB disk/project
445GB consumed by projects with more than 1GB but less than 10GB disk/project
1110GB consumed by projects with more than 10GB disk/project

**Comment by wmhutchison** [comment]
Currently waiting on a follow-up audit to complete after being informed by devs that some clean-up has taken place. Will post the results here.

**Comment by lukegonis** [comment]
no longer alerting anymore, moving to closed, a new ticket to explore automation options in the OCP4 board.
